### DISCUSSION TITLE
How to show a user that he is banned?

### DISCUSSION THREAD

**Discussion by ioancruso** [supabase/supabase#22775] [Questions]
Hello! When banning someone using banned_until in auth.users, when they try to login they receive the error:

`AuthApiError: Invalid login credentials
  __isAuthError: true,
  status: 400,
  code: undefined
}`

The things is I want to tell them that they are banned. Is there any way to make login returns something else when the user is banned?

**Comment by reid-moffat** [comment]


**Reply by alexcdot** [reply]
I don't see the banned_until field in the docs nor do i see it when i test it on my own server https://supabase.com/docs/reference/javascript/auth-admin-getuserbyid

**Comment by joseppu** [comment]
It is done in this way. it logouts user, doesn't let user refresh token. throws the error below. also check the github comment below for more info.

```js

 const { data, error } = await supabase2.auth.admin.updateUserById(userId, {
 ban_duration: "5m", }
);
```

```
Error [AuthApiError]: Invalid Refresh Token: User Banned
    at handleError (../../../src/lib/fetch.ts:102:8)
    at async _handleRequest (../../../src/lib/fetch.ts:195:4)
    at async _request (../../../src/lib/fetch.ts:157:15)
    at async (../../src/GoTrueClient.ts:2112:17)
    at async (../../../src/lib/helpers.ts:228:25) {
  __isAuthError: true,
  status: 400,
  code: 'user_banned'
}
```


https://github.com/orgs/supabase/discussions/9239#discussioncomment-3766124
